librz/il/definitions/bitvector.c

```diff
@@ -0,0 +1,1082 @@
+	// dump bv to a new one
+	if (!bv || !(bv->bits)) {
+		return NULL;
+	}
+
+	BitVector new_bv = rz_il_bv_new(bv->len);
+	if (!new_bv || !(new_bv->bits)) {
+		return NULL;
+	}
+
+	for (int i = 0; i < bv->_elem_len; ++i) {
+		new_bv->bits[i] = bv->bits[i];
+	}
+
+	return new_bv;
+}
+
+/**
+ * copy from source bitvector to destination bitvector
+ * the maximum copied size depends on MIN(src_len, dst_len)
+ * \param src BitVector, the source bitvector
+ * \param dst BitVector, the destination bitvector
+ * \return Actual size of copy
+ */
+RZ_API int rz_il_bv_copy(BitVector src, BitVector dst) {
+	if (!dst || !(dst->bits) || !src || !(src->bits)) {
+		return 0;
+	}
+
+	if (dst->len != src->len) {
+		return 0;
+	}
+
+	for (int i = 0; i < dst->_elem_len; ++i) {
+		dst->bits[i] = src->bits[i];
+	}
+
+	return dst->_elem_len;
+}
+
+/**
+ * Copy n bits from start position of source to start position of dest
+ * \param src BitVector, data source
+ * \param src_start_pos int, start position in source bitvector of copy
+ * \param dst BitVector, destination of copy
+ * \param dst_start_pos int, start position in destination bitvector
+ * \param nbit int, control the size of copy (in bits)
+ * \return copied_size int, Actual copied size
+ */
+RZ_API int rz_il_bv_copy_nbits(
+	BitVector src, int src_start_pos,
+	BitVector dst, int dst_start_pos,
+	int nbit) {
+	if (!dst || !(dst->bits) || !src || !(src->bits)) {
+		return 0;
+	}
+
+	int max_nbit = RZ_MIN((src->len - src_start_pos),
+		(dst->len - dst_start_pos));
+
+	// prevent overflow
+	if (max_nbit < nbit) {
+		return 0;
+	}
+
+	// normal case here
+	for (int i = 0; i < max_nbit; ++i) {
+		bit c = rz_il_bv_get(src, src_start_pos + i);
+		rz_il_bv_set(dst, dst_start_pos + i, c);
+	}
+
+	return nbit;
+}
+
+/**
+ * Adjust bitvector to a new length
+ * TODO : remove this and replace with append/prepend/cut_head/cut_tail
+ * \param bv BitVector, pointer to bv need adjustment
+ * \param new_len int, target length
+ * \return ret BitVector, pointer to the new bitvector
+ */
+BitVector bv_adjust(BitVector bv, int new_len) {
+	if (!bv || !bv->bits) {
+		return NULL;
+	}
+
+	BitVector ret = rz_il_bv_new(new_len);
+	if (ret == NULL) {
+		return NULL;
+	}
+
+	int max_len = RZ_MIN(ret->_elem_len, bv->_elem_len);
+	for (int i = 0; i < max_len; ++i) {
+		ret->bits[i] = bv->bits[i];
+	}
+	return ret;
+}
+
+/**
+ * Prepend bv with n zero bits
+ * \param bv BitVector, pointer to bitvector instance
+ * \param delta_len int, the number of zero bits
+ * \return ret BitVector, pointer to the new bitvector instance
+ */
+RZ_API BitVector rz_il_bv_prepend_zero(BitVector bv, int delta_len) {
+	if (!bv || !bv->bits) {
+		return NULL;
+	}
+
+	int new_len = bv->len + delta_len;
+	BitVector ret = rz_il_bv_new(new_len);
+	if (ret == NULL) {
+		return NULL;
+	}
+
+	int pos = delta_len;
+	for (int i = 0; i < bv->len; ++i, ++pos) {
+		rz_il_bv_set(ret, pos, rz_il_bv_get(bv, i));
+	}
+	return ret;
+}
+
+/**
+ * Append bv with n zero bits
```

Comment:
As I see it returns NEW bitvector

Reply:
Indeed it is, I will change the description